types/index.d.ts

```diff
@@ -739,6 +739,32 @@ declare namespace Knex {
       column: string | readonly string[]
     ): QueryBuilder<TRecord, TResult2>;
 
+    onConflict<
+      TKey extends StrKey<TRecord>,
+      TResult2 = DeferredIndex.Augment<
+        UnwrapArrayMember<TResult>,
+        TRecord,
+        TKey
+      >[]
+    >(
+      column: TKey
+    ): QueryBuilder<TRecord, TResult2>;
+    onConflict<
+      TKey extends StrKey<TRecord>,
+      TResult2 = DeferredKeySelection.SetSingle<
+        DeferredKeySelection.Augment<UnwrapArrayMember<TResult>, TRecord, TKey>,
+        false
+      >[]
+    >(
+      columns: readonly TKey[]
+    ): QueryBuilder<TRecord, TResult2>;
+    returning<TResult2 = SafePartial<TRecord>[]>(
+      column: string | readonly string[]
+    ): QueryBuilder<TRecord, TResult2>;
+
+    ignore(): QueryBuilder<TRecord, TResult>;
+    merge(data?: DbRecordArr<TRecord>): QueryBuilder<TRecord, TResult>;
```

Comment:
I think here should be `DbRecord` instead of `DbRecordArr`

Reply:
Ah yeah, that sounds right. This function doesn't accept an array.